src/server/routes.rs

```diff
@@ -71,6 +73,22 @@ pub mod v1 {
                 .and(warp::post())
                 .and(with_store(store))
                 .and(filters::toml())
+                .and(warp::header::optional::<String>("accept"))
+                .and_then(create_invoice)
+                .recover(filters::handle_deserialize_rejection)
+        }
+        pub fn create_json<P>(
+            store: P,
+        ) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone
+        where
+            P: Provider + Clone + Send + Sync,
+        {
+            warp::path("_i")
+                .and(warp::path::end())
+                .and(warp::post())
+                .and(with_store(store))
+                .and(filters::json())
+                .and(warp::header::optional::<String>("accept"))
```

Comment:
All these `accept` header passing things are...not fun (I think this was the stuff you mentioned offline), but I really don't think there is any other way to do this